lib/yabeda/puma/plugin/statistics/fetcher.rb

```diff
@@ -7,7 +7,18 @@ module Plugin
       module Statistics
         class Fetcher
           def self.call
-            body = Socket.unix(Yabeda::Puma::Plugin.control_url.gsub("unix://", '')) do |socket|
+            control_url = Yabeda::Puma::Plugin.control_url
+
+            if control_url.include? "unix://"
+              path = control_url.gsub("unix://", '')
+              sock = Socket.unix(path)
+            elsif control_url.include? "tcp://"
+              host, port = control_url.match(/^tcp:\/\/([a-z0-9\-._~%]+):([0-9]+)/).captures
+
+              sock = Socket.tcp(host, port)
+            end
```

Comment:
hey! Great job! 💪
To keep things simple let just check prefix of the `control_url`.
And also raise an error in unexpected case, to prevent the unclear behavour you've met before (when code silently does not expect some kind of value).

```ruby
if control_url.start_with?('unix://')
  # do unix socket
elsif control_url.start_with?("tpc://")
  # do tcp socket
else
  raise ArgumentError("Unknown puma control url type #{control_url}")
end
```

Reply:
Hey thank you :-) I made the changes you requested.